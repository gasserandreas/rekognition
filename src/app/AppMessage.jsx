import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import ViewMessage from '../ui/ViewMessage';
import Button from '../ui/form/Button';
import ButtonGroup from '../ui/form/ButtonGroup';

const AppMessage = ({
  message: {
    text, show, title, showRefresh,
  },
}) => (
  <ViewMessage message={text} show={show}>
    <h1>{title}</h1>
    <p>{text}</p>
    {showRefresh && (
      <Fragment>
        <p>Please refresh page and try again. If error persists please try later.</p>
        <ButtonGroup>
          <Button type="button" buttonStyle="primary" onClick={() => window.location.reload()}>
            Refresh page
          </Button>
        </ButtonGroup>
      </Fragment>
    )}
  </ViewMessage>
);

AppMessage.propTypes = {
  message: PropTypes.shape({
    show: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    showRefresh: PropTypes.bool.isRequired,
  }).isRequired,
};

export default AppMessage;
