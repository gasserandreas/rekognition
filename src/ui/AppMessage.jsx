import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Box } from 'grommet';

import { Colors, MediaSize } from '../styles';

export const StyledAppMessage = styled(Box)`
  visibility: ${props => props.show ? 'visible' : 'none'};
  display: ${props => props.show ? 'block' : 'none' };
  position: fixed;

  /* positioning */
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  height: 100%;
  z-index: 1000;

  @media (min-width: ${MediaSize.Tablet}) {
    background-color: rgba(48, 48, 48, 0.4);
  }
`;

export const StyledAppMessageContent = styled(Box)`
  background-color: ${Colors.ColorsPalette.White};
  transition: top ease-in-out 500ms;
  padding: 2rem;

  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 100%;
  
  @media (min-width: ${MediaSize.Tablet}) {
    top: ${props => props.show ? '10rem' : '-100%'};
    height: auto;
    width: 32rem;
    right: auto;
    left: 50%;
    margin-left: -16rem;
    padding: 1rem;
    min-height: 10rem;
    max-height: 30rem;
  }

  h1, h2, h3, h4, h5 {
    text-transform: uppercase;
    font-size: 1.25rem;
    font-weight: 500;
  }

  p {
    margin: 0.5rem 0;
  }
`;

const AppMessage = ({ children, show, ...props }) => (<StyledAppMessage show={show} {...props}>
    <StyledAppMessageContent elevation="xlarge" show={show}>
      {children}
    </StyledAppMessageContent>
  </StyledAppMessage>
);

AppMessage.propTypes = {
  children: PropTypes.node.isRequired,
  show: PropTypes.bool,
}

AppMessage.defaultProps = {
  show: false,
}

export default AppMessage;
