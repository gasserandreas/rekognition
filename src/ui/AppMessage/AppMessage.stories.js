import React from 'react';

import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';

import AppMessage from './AppMessage';

storiesOf('AppMessage', module)
  .add('with customized info', () => (
    <AppMessage show={boolean('show', true)}>
      <h1>AppMessage Example</h1>
      <p>Add some text in here and...</p>
    </AppMessage>
  ))
  // ), {
  //   info: {
  //     text: 'hello world',
  //   },
  // })
  // ), {
  //   info: {
  //     text: Markdown,
  //   },
  // })
  .add('with content only', () => (
    <AppMessage show={boolean('show', true)}>
      <h1>AppMessage Example</h1>
      <p>Add some text in here and...</p>
    </AppMessage>
  ));
