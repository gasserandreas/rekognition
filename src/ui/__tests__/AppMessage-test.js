/* global React, shallow, mount, render, renderer, testSnapshot */

import AppMessage, { StyledAppMessage, StyledAppMessageContent } from '../AppMessage';

// test styling
it('AppMessage should render correctly', () => {
  testSnapshot(<AppMessage><p>Hello</p></AppMessage>);
});

it('AppMessage should be styled consistently', () => {
  testSnapshot(<StyledAppMessage />);
});

it('AppMessage content should be styled consistently', () => {
  testSnapshot(<StyledAppMessageContent />);
});

it('AppMessage content should be visible', () => {
  testSnapshot(<AppMessage show={true}><p>Hello</p></AppMessage>);
})

// unit tests
it('AppMessage content should show/hide on prop change', () => {
  const output = mount(<AppMessage><p>Hello</p></AppMessage>);
  expect(output);
  expect(output.props().show).toEqual(false);
  
  // check for content
  let content = output.find(StyledAppMessageContent);
  expect(content);
  expect(content.props().show).toEqual(false);

  // should be visible
  output.setProps({ show: true });
  expect(output.props().show).toEqual(true);
  content = output.find(StyledAppMessageContent);
  expect(content.props().show).toEqual(true);

  // should be hidden again
  output.setProps({ show: false });
  expect(output.props().show).toEqual(false);
  content = output.find(StyledAppMessageContent);
  expect(content.props().show).toEqual(false);
});
