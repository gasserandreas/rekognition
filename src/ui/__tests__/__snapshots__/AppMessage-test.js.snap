// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMessage content should be styled consistently 1`] = `
<div
  className="sc-bwzfXH fswprr StyledBox-sc-13pk1d4-0 IRSNj"
/>
`;

exports[`AppMessage content should be visible 1`] = `
<div
  className="sc-bdVaJa knjkYo StyledBox-sc-13pk1d4-0 IRSNj"
>
  <div
    className="sc-bwzfXH jCUgKl StyledBox-sc-13pk1d4-0 hEFsDQ"
  >
    <p>
      Hello
    </p>
  </div>
</div>
`;

exports[`AppMessage should be styled consistently 1`] = `
<div
  className="sc-bdVaJa jVkiAG StyledBox-sc-13pk1d4-0 IRSNj"
/>
`;

exports[`AppMessage should render correctly 1`] = `
<div
  className="sc-bdVaJa jVkiAG StyledBox-sc-13pk1d4-0 IRSNj"
>
  <div
    className="sc-bwzfXH fswprr StyledBox-sc-13pk1d4-0 hEFsDQ"
  >
    <p>
      Hello
    </p>
  </div>
</div>
`;
